### PR TITLE
libvirt.py: add extra patterns to connect libvirtd

### PR DESCRIPTION
**Pull request by dzhengfy**
Existing connect_libvirtd function has limited patterns to match in the
command output. Those patterns can not satisfy all demands. The fix
provides a new mapping including the patterns and corresponding
responses, which makes the application which calls this function
specify the patterns and responses it needs.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
This is already tested locally on RHEL7.2 with libvirt-1.3.2-1.el7.x86_64.
JOB ID     : d2600be25ee9f2d25452c023b0df34f9c03da121
JOB LOG    : /root/avocado/job-results/job-2016-04-01T15.42-d2600be/job.log
TESTS      : 11
 (1/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.allow_one_sasl_user: PASS (20.79 s)
 (2/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.allow_sasl_users: PASS (26.39 s)
 (3/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv4_tcp_port: PASS (18.85 s)
 (4/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv4_listen_address_with_sasl_auth: PASS (25.23 s)
 (5/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv6_listen_address: PASS (24.46 s)
 (6/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv6_tcp_port: PASS (23.92 s)
 (7/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.positive_testing.customized_ipv6_listen_address_with_sasl_auth: PASS (29.23 s)
 (8/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.negative_testing.only_simpy_connect_libvirtd: PASS (15.89 s)
 (9/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.negative_testing.no_allowed_sasl_user: PASS (22.34 s)
 (10/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.negative_testing.ipv4_with_allowed_sasl_user_readonly: PASS (24.55 s)
 (11/11) type_specific.io-github-autotest-libvirt.virsh.remote_with_tcp.negative_testing.ipv6_with_readonly: PASS (26.56 s)

JOB LOG    : /root/avocado/job-results/job-2016-04-01T15.51-a0a5114/job.log
TESTS      : 26
 (1/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.default_tls_port_and_auth_tls: PASS (27.82 s)
 (2/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.no_verify_certificate: PASS (31.67 s)
 (3/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.no_sanity_checks_and_no_verify: PASS (35.87 s)
 (4/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_tls_port: PASS (30.77 s)
 (5/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.allow_one_sasl_user: PASS (36.85 s)
 (6/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.allow_sasl_users: PASS (38.76 s)
 (7/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.allowed_dn_list: PASS (35.55 s)
 (8/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_ipv4_listen_address: PASS (28.89 s)
 (9/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_ipv4_listen_address_with_sasl_auth: PASS (37.50 s)
 (10/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_ipv6_listen_address: PASS (37.56 s)
 (11/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_ipv6_tls_port: PASS (44.66 s)
 (12/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_ipv6_listen_address_with_sasl_auth: PASS (45.55 s)
 (13/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.positive_testing.customized_x509_cert_path: PASS (32.15 s)
 (14/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.only_simpy_connect_libvirtd: PASS (15.71 s)
 (15/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.invalid_tls_port: PASS (30.55 s)
 (16/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.tls_port_in_use: PASS (31.03 s)
 (17/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.tls_listen_addr_in_use: PASS (29.97 s)
 (18/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.ipv4_with_allowed_sasl_user_readonly: PASS (39.01 s)
 (19/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.no_sanity_checks_without_no_verify: PASS (36.66 s)
 (20/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.allowed_dn_disorder_list: PASS (33.26 s)
 (21/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.allowed_dn_list_with_invalid_organization: PASS (30.10 s)
 (22/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.allowed_dn_list_with_invalid_CN: PASS (32.15 s)
 (23/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.verify_certificate: PASS (30.89 s)
 (24/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.incorrect_ipv4_listen_address: PASS (31.94 s)
 (25/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.ipv6_with_readonly: PASS (38.33 s)
 (26/26) type_specific.io-github-autotest-libvirt.virsh.remote_with_tls.negative_testing.ipv6_address_unreachable: PASS (35.53 s)
JOB LOG    : /root/avocado/job-results/job-2016-04-01T10.16-85c9282/job.log
TESTS      : 17
 (1/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.ssh_diff_libvirt_version: SKIP
 (2/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.ssh_static_ipv6: PASS (32.26 s)
 (3/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.gssapi_auth: PASS (17.53 s)
 (4/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.test_uri_with_default_user: PASS (17.44 s)
 (5/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.test_uri_with_root_user: PASS (16.67 s)
 (6/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.xen_uri_with_default_user: PASS (15.24 s)
 (7/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.xen_uri_with_root_user: PASS (14.27 s)
 (8/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.esx_uri_with_default_user: PASS (16.42 s)
 (9/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.positive_testing.esx_uri_with_root_user: PASS (21.43 s)
 (10/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.libssh2_no_auth: PASS (16.43 s)
 (11/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.ssh_no_uri_path: PASS (24.37 s)
 (12/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.ssh_read_only_mode: PASS
 (13/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.ssh_no_ipv6_config: PASS (23.34 s)
 (14/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.stop_libvirtd: PASS (24.45 s)
 (15/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.incorrect_password: PASS (20.05 s)
 (16/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.ipv4_with_readonly: PASS
 (17/17) type_specific.io-github-autotest-libvirt.virsh.remote_with_ssh.negative_testing.ipv6_with_readonly: PASS

**Comment by will-Do** [comment]
I'm OK with this patch, please fix the Travis CI failure

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -2473,9 +2473,38 @@ def remotely_control_libvirtd(server_ip, server_user, server_pwd,
...
-                     patterns_virsh_cmd=".*Id\s*Name\s*State\s*.*"):
-    """
-    Connect libvirt daemon
+                     patterns_virsh_cmd=".*Id\s*Name\s*State\s*.*",
+                     patterns_extra_dict=None):
+    """
+    Connect to libvirt daemon
+
+    :param uri: the uri to connect the libvirtd
+    :type str
```
Comment (waynesun09) [inline virttest/utils_test/libvirt.py]: The docstring are not right and cause ci failure
s/:type/:type uri:/

**Comment by dzhengfy** [comment]
@will-Do @waynesun09 , I removed :type line for docstring

**Comment by will-Do** [comment]
Okay then, I will merge this pr after get another ACK

**Comment by waynesun09** [comment]
ACK

**Pull request merged**